src/Api/SecretsManager/Controllers/SecretsManagerPortingController.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Bit.Api.SecretsManager.Models.Request;
+using Bit.Api.SecretsManager.Models.Response;
+using Bit.Core.Context;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
+using Bit.Core.SecretsManager.Commands.Porting.Interfaces;
+using Bit.Core.SecretsManager.Repositories;
+using Bit.Core.Services;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Bit.Api.SecretsManager.Controllers;
+
+[SecretsManager]
+public class SecretsManagerPortingController : Controller
+{
+    private readonly ISecretRepository _secretRepository;
+    private readonly IProjectRepository _projectRepository;
+    private readonly IUserService _userService;
+    private readonly IImportCommand _importCommand;
+    private readonly ICurrentContext _currentContext;
+
+    public SecretsManagerPortingController(ISecretRepository secretRepository, IProjectRepository projectRepository, IUserService userService, IImportCommand importCommand, ICurrentContext currentContext)
+    {
+        _secretRepository = secretRepository;
+        _projectRepository = projectRepository;
+        _userService = userService;
+        _importCommand = importCommand;
+        _currentContext = currentContext;
+    }
+
+    [HttpGet("sm/{organizationId}/export")]
+    public async Task<SMExportResponseModel> Export([FromRoute] Guid organizationId, [FromRoute] string format = "json")
+    {
+        var userId = _userService.GetProperUserId(User).Value;
+        var orgAdmin = await _currentContext.OrganizationAdmin(organizationId);
+        var accessClient = AccessClientHelper.ToAccessClient(_currentContext.ClientType, orgAdmin);
+
+        var projects = await _projectRepository.GetManyByOrganizationIdAsync(organizationId, userId, accessClient);
+        var secrets = await _secretRepository.GetManyByOrganizationIdAsync(organizationId); // TODO: add access check for secrets here once it's supported
```

Comment:
Access checks for getting secrets is implemented (PR open, not merged yet) so that the SecretsController GetSecretsByOrganizationAsync function will do the check. I was recommended to not have the secretsRepo call the projectRepo (required for access checks on secrets), so I'm not sure how we can get around this since you are calling the secretsRepo directly instead of going through the controller. 🤔

Reply:
We should just do an admin check at the beginning of this method, since it's limited to admins.

Reply:
perfect, thanks Oscar!!

Reply:
Thanks for mentioning this, updated!